### ISSUE TITLE
Error output (in flat config) causes problems not to appear in the output channel

### ISSUE THREAD

**Issue by brettz9** [microsoft/vscode-eslint#1894]
With flat config at least, it seems that the presence of warnings on the command line prevent vscode from reporting anything at all.

To reproduce:

1. Add this to a flat config file (`eslint.config.js`):

```js
// Note the missing semicolon which should trigger an error
import noUseExtendNative from 'eslint-plugin-no-use-extend-native'

export default [
  {
    rules: {
      semi: ['error']
    }
  }
];
```

2. Notice after a restart there are no ESLint error messages showing up when the output channel is shown.
3. Run `eslint .` from a `package.json` script and notice the deprecation error due to the above import currently giving such an error.
4. Change the `eslint.config.js` file to the following:

```js
// Missing semi at the end
export default [
  {
    rules: {
      semi: ['error']
    }
  }
]
```
5. Notice after a restart that the error shows in VSCode under `problems` when the output channel is shown.

**Comment by dbaeumer** [comment]
@brettz9 since something like this is very version dependent (both eslint and plugins) would you be able to provide me with a GitHub repository wit a minimal setup to reproduce the problem.

**Comment by brettz9** [comment]
Apologies, after setting up the repo, I noticed that there was in fact output, just no problems.

And that was due to using different Node versions between command line and in VSC, something resolved by https://github.com/microsoft/vscode-eslint/issues/1356#issuecomment-1283776109 .

Thanks for the help!
